src/blueprint/from_config/operators/call.rs

```diff
@@ -0,0 +1,59 @@
+use crate::blueprint::*;
+use crate::config;
+use crate::config::{Config, Field, GraphQLOperationType};
+use crate::directive::DirectiveCodec;
+use crate::try_fold::TryFold;
+use crate::valid::Valid;
+
+pub fn update_call(
+  operation_type: &GraphQLOperationType,
+) -> TryFold<'_, (&Config, &Field, &config::Type, &str), FieldDefinition, String> {
+  TryFold::<(&Config, &Field, &config::Type, &str), FieldDefinition, String>::new(
+    move |(config, field, type_of, name), b_field| {
+      let Some(call) = &field.call else {
+        return Valid::succeed(b_field);
+      };
+
+      let type_and_field = if let Some(mutation) = &call.mutation {
+        Valid::succeed(("Mutation", mutation.as_str()))
+      } else if let Some(query) = &call.query {
+        Valid::succeed(("Query", query.as_str()))
+      } else {
+        Valid::fail("call must have one of mutation or query".to_string())
+      };
+
+      type_and_field
+        .and_then(|(type_name, field_name)| {
+          Valid::from_option(
+            config.find_type(type_name),
+            format!("{} type not found on config", type_name),
+          )
+          .zip(Valid::succeed(field_name))
+        })
+        .and_then(|(query_type, field_name)| {
+          Valid::from_option(
+            query_type.fields.get(field_name),
+            format!("{} field not found", field_name),
+          )
+          .and_then(|field| {
+            if !field.has_resolver() {
+              return Valid::fail(format!("{} field has no resolver", field_name));
+            }
+
+            Valid::succeed(field)
+          })
+        })
+        .and_then(|field| {
+          // TO-DO: parse call.args into a way that `update_http`, `update_grpc` and `update_graphql` can use
```

Comment:
can you give a hand on it @tusharmath @amitksingh1490 @meskill?

IMO, most of it is done (besides automated testing), but I need help with the args.

What I have in mind now is to search based on the `field.args` mustache value. So if I have `value.userId`, I'd search for a type with a `userId` and the `field_name` `user` using this `@call` operator. Then I get the type from `userId` and use it to create an `Arg` that'll be set on `b_field`, but I'm not sure if this is the best approach.

Reply:
got it working :smile: